airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -179,18 +183,30 @@ private void runInternal(final IntegrationConfig parsed) throws Exception {
     LOGGER.info("Completed integration: {}", integration.getClass().getName());
   }
 
-  private boolean isConfigError(final Exception e) {
+  private Throwable getRootThrowable(final Exception e) {
+    Throwable current = e;
+    while (current != null) {
+      if (isConfigError(current)) {
+        return current;
+      } else {
+        current = current.getCause();
```

Comment:
Like peeling an onion..

Reply:
May be able to simple use google common's library to do something like this

https://github.com/spotify/helios/blob/0b2855955dc6f3268e772252d84861082a516b6e/helios-services/src/main/java/com/spotify/helios/agent/HealthCheckerFactory.java#L153-L159

Reply:
Good find! Unfortunately, `Throwables.getRootCause` gets the root exception. Here, we actually want the root exception that is a possible configexception. The nuance here is that we *could* catch some `SQLException`, repackage it as a `ConfigErrorException` which then is repackaged up the stack as a `RuntimeException`.

We actually want the `ConfigErrorException`, so I think we have to manually unnest this

Reply:
I see, then can we add a javadoc comment that indicates this looks for the first instance of a `ConfigErrorException`? Also curious what happens if the Exception does not contain a `ConfigErrorException`, it seems would it be returning `null`, right?

Reply:
Ahhh good catch :(
Will just return the original exception if there is no config error. Also updated the comments